### PR TITLE
CnCNetLobby message right click context menu

### PR DESCRIPTION
**Pull request by devo1929**
Right click context menu for player-sent messages in the cncnet main lobby.

For now, this uses the existing functionality of the current player right-click context menu. However, it is a derived class leaving it open to other future menu options.

### CODE REVIEW

**Comment by devo1929** [comment]
image

**Comment by Metadorius** [comment]
Would it be possible to add a "Copy text" menu entry and (when a link is detected) "Open link" and "Copy link"?

File: DXMainClient/DXGUI/Multiplayer/ChatListBox.cs [hunk 1]
```diff
@@ -52,17 +52,22 @@ public void AddMessage(string sender, string message, Color color)
...
-                    message.Color, true);
+                    message.Message), FontIndex);
             else
-            {
-                AddItem(Renderer.GetSafeString(string.Format("[{0}] {1}: {2}",
-                    message.DateTime.ToShortTimeString(), message.SenderName, message.Message), FontIndex),
-                    message.Color, true);
-            }
+                listBoxItem.Text = Renderer.GetSafeString(string.Format("[{0}] {1}: {2}",
+                    message.DateTime.ToShortTimeString(), message.SenderName, message.Message), FontIndex);
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Multiplayer/ChatListBox.cs:68]: Any multi-line code blocks should use braces for better readability and excluding the possibility of [goto fail error](https://www.synopsys.com/blogs/software-security/understanding-apple-goto-fail-vulnerability-2/).

**Review by Metadorius — CHANGES_REQUESTED** [review]
Looks good aside from a braces issue.

**Comment by devo1929** [comment]
> Would it be possible to add a "Copy text" menu entry and (when a link is detected) "Open link" and "Copy link"?

I like this idea, but maybe next PR? That'll include some parsing work.

File: DXMainClient/DXGUI/Multiplayer/ChatListBox.cs (hunk 1 above)
Comment (devo1929) [reply DXMainClient/DXGUI/Multiplayer/ChatListBox.cs]: Fixed.

**Comment by Metadorius** [comment]
Links are already parsed in code, try double-clicking any message with a link. It's just not highlighted due to how XNAUI works with text. Look for `http://` and `https://` in code, should be trivial to patch that into a menu entry.

**Comment by devo1929** [comment]
> Links are already parsed in code, try double-clicking any message with a link. It's just not highlighted due to how XNAUI works with text. Look for `http://` and `https://` in code, should be trivial to patch that into a menu entry.

I found the code. I started to implement it and it works, but right now, the menu is in the context of a "player/user". It won't show if there's no user associated with the message you're right-clicking on. System messages (like the IP ban one) won't have a context menu. It's not a difficult add, but it's not insignificant either.

**Comment by Metadorius** [comment]
Then I guess feel free to leave that for another PR 🤷‍♂️

**Review by Metadorius — APPROVED** [review]

**Pull request merged**